lib/rtsp.c

```diff
@@ -822,6 +861,49 @@ CURLcode Curl_rtsp_parseheader(struct Curl_easy *data, char *header)
       (data->set.str[STRING_RTSP_SESSION_ID])[idlen] = '\0';
     }
   }
+  else if(checkprefix("Transport:", header)) {
+    /* e.g.: 'Transport: RTP/AVP/TCP;unicast;interleaved=5-6' */
+    char *start;
+    char *end;
+    start = header + 10;
+    while(*start) {
+      while(*start && *start != ';' && ISBLANK(*start) )
+        start++;
+      end = start;
+      while(*end && *end != ';')
+        end++;
+      if(checkprefix("interleaved=", start)) {
+        long chan1, chan2, chan;
+        char *endp;
+        char *p = start + 12;
+        chan1 = strtol(p, &endp, 10);
+        if(p != endp && chan1 >= 0 && chan1 <= 255) {
+          unsigned char *rtp_channel_mask = data->state.rtp_channel_mask;
+          chan2 = chan1;
+          if(*endp == '-') {
+            p = endp + 1;
+            chan2 = strtol(p, &endp, 10);
+            if(p == endp || chan2 < 0 || chan2 > 255) {
+              failf(data, "Unable to read the interleaved parameter from "
+                    "Transport header: [%s]", header);
+              chan2 = chan1;
+            }
+          }
+          for(chan = chan1; chan <= chan2; chan++) {
+            long idx = chan / 8;
+            long off = chan % 8;
+            rtp_channel_mask[idx] |= (unsigned char)(1 << off);
+          }
+        }
+        else {
+          failf(data, "Unable to read the interleaved parameter from "
+                "Transport header: [%s]", header);
+        }
+      }
+      /* skip to next parameter */
+      start = (!*end) ? end : (end + 1);
```

Comment:
Can there be more than one `interleaved=` within the same header?

Can there be more than one `Transport:` header for the same content?

Reply:
Only one transport specification can be included in the Transport response-header, which contains multiple parameters separated by semicolons, which will contain only one interleaved parameter.  However, loop through all parameters can standardize the processing, and easily add parsing of other parameters.

I think one RTSP message can only contain one Transport header but there can be multiple SETUP requests to control different streams separately.